assets/pages/new-contributions-landing/components/ContributionState.jsx

```diff
@@ -36,15 +44,22 @@ const renderStatesField = states => (
   </div>
 );
 
-function NewContributionState(props: PropTypes) {
+function ContributionState(props: PropTypes) {
   switch (props.countryGroupId) {
     case 'UnitedStates':
-      return renderStatesField(usStates);
+      return renderStatesField(usStates, props.onChange);
     case 'Canada':
-      return renderStatesField(caStates);
+      return renderStatesField(caStates, props.onChange);
     default:
       return null;
   }
 }
 
+
+ContributionState.defaultProps = {
+  onChange: false,
```

Comment:
This seems odd to me (although I'm not saying that's because it's wrong!). The pattern I have used in the past in this codebase when needing a default prop for an `onChange` would be to do something like this:

`onChange: () => undefined`

It just seems strange to have on change being either a boolean or a function. What do you think?

Reply:
This is interesting. Flow marks `onChange` on the [corresponding DOM element](https://github.com/facebook/flow/blob/v0.79.1/lib/dom.js#L1517) as `?Function` (browser API, not React). This does seem odd in terms of types, but then again it may save unnecessary function calls (or it may not, if empty functions like this get optimised away).

I don't feel too strongly, although it occurs to me that potentially `(Event => void) | void` may also work, which will allow you to use `undefined` as the default prop. Since this gets [turned into](https://reactjs.org/docs/introducing-jsx.html#jsx-represents-objects) `React.createElement('select', { onChange: undefined })` it should be the equivalent to the prop not existing.